### PR TITLE
Split up remaining pieces in public/index.php

### LINKED ISSUE #4: Extract further closures contained in `public/index.php` into independent units
The `index.php` entry-point contains a bunch of on-the-fly generated closures that aren't covered by tests:

https://github.com/doctrine/automatic-releases/blob/8b945ef4976d28067e0f737b5c99862640ef5260/public/index.php#L55-L140

These should be extracted to classes, and tested in isolation. Methods that perform I/O via CLI need an integration test.

### PR DESCRIPTION
**Pull request by jwage**
Fixes #4

TODO:

- [ ] get feedback from Marco on direction
- [ ] I think maybe a few other things could be split out a little more
- [ ] Add tests

### CODE REVIEW

File: src/Git/CleanBuildDir.php [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+
+namespace Doctrine\AutomaticReleases\Git;
+
+use Symfony\Component\Process\Process;
+
+final class CleanBuildDir
+{
+    public function __invoke(string $buildDir) : void
+    {
+        (new Process(['rm', '-rf', $buildDir]))
```
Comment (Ocramius) [inline src/Git/CleanBuildDir.php:13]: At this point, we need to assert that `$buildDir` is a directory, before performing something this scary :)

File: src/Git/CloneRepository.php [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+
+use Psr\Http\Message\UriInterface;
+use Symfony\Component\Process\Process;
+
+final class CloneRepository
+{
+    public function __invoke(
+        UriInterface $repositoryUri,
+        string $targetPath,
+        string $gitAuthorName,
```
Comment (Ocramius) [inline src/Git/CloneRepository.php:15]: These values should be asserted upon, or we should have value types introduced

File: src/Git/CreateTag.php [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+use function Safe\file_put_contents;
+use function Safe\tempnam;
+use function sys_get_temp_dir;
+
+final class CreateTag
+{
+    public function __invoke(
+        string $repositoryDirectory,
+        BranchName $sourceBranch,
+        string $tagName,
```
Comment (Ocramius) [inline src/Git/CreateTag.php:19]: We need to assert on the tag being a non-empty string

**Review by Ocramius — CHANGES_REQUESTED** [review]

**Comment by Ocramius** [comment]
> get feedback from Marco on direction

As you can see from the structure of the code that was ported, it is extremely hard to test in isolation. That will be a pain, but we'll need to go through it. Also, I think that the various `string` parameters, especially when representing a `Directory` or a `GitRepositoryRootPath`, we should have value types representing them, reducing the risk of incorrect strings being passed around.

**Comment by jwage** [comment]
Cool. That is aligned with what I was thinking needs to happen next. I will keep working on it.

**Comment by Ocramius** [comment]
Closing here: this has been implemented in https://github.com/laminas/automatic-releases/pull/1, which splits the application into multiple independent CLI commands.

**Pull request closed without merging**